### ISSUE TITLE
determine best framework for errors and validation

### ISSUE THREAD

**Issue by benguhin** [thepolicylab/covid-forms#7]
breaking out from #5

**Comment by theresaanna** [comment]
Research so far:

I am looking at three different libraries. 
https://parsleyjs.org/
https://validatejs.org/
https://formvalidation.io/

Parsley looks like a simpler basic implementation. Validate looks most flexible. FormValidation has some nice built in validators and plugin system. Each are well-documented. Parsley looks to, based on examples, require extra work to do certain things, like immediate feedback to the user if there's a validation error. You do need to click the submit button before receiving feedback, by default, in Parsley, and I don't see examples that do it automatically. @benguhin do you have any thoughts about whether there's an optimal path here? Clicking submit vs. real-time feedback? Here are the example forms: https://parsleyjs.org/doc/examples/simple.html https://validatejs.org/examples.html https://formvalidation.io/guide/examples

FormValidation gives a lot of great examples, which I think would make implementation easiest while giving a lot of power. I am leaning in that direction but would love thoughts!

**Comment by khwilson** [comment]
Not my call on design, but I'm a fan of realtime feedback and _client side_ validation whenever possible. :-)

**Comment by benguhin** [comment]
Yeah I think real-time client-side feedback would be great. Any reason to not use FormValidation? (I will go ahead and say that the license cost is not a blocker)

**Comment by theresaanna** [comment]
@benguhin No, I think FormValidation might be the best bet. Do you know what we'd have to do to get a license?

**Comment by theresaanna** [comment]
I'm going to close out this issue because we've determined the best framework. Now we need to figure out how to purchase and implement, which is a separate issue.

**Comment by theresaanna** [comment]
Further work needed: research HTML5 built in form validation with our form and see if it does what is needed. Jumping off point: https://developer.mozilla.org/en-US/docs/Learn/Forms/HTML5_input_types
